server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -811,7 +834,23 @@ public VectorData parseKnnVector(
             XContentParser.Token token = context.parser().currentToken();
             return switch (token) {
                 case START_ARRAY -> parseVectorArray(context, dims, dimChecker, similarity);
-                case VALUE_STRING -> parseHexEncodedVector(context, dimChecker, similarity);
+                case VALUE_EMBEDDED_OBJECT -> parseBase64BinaryEncodedVector(context.parser().binaryValue(), dimChecker, similarity);
```

Comment:
I guess this is for CBOR / SMILE? Do we have tests that checks this?

Reply:
>Switching from base64 to "Nah, just send bytes" is pretty inconsistent no?

It is the same inconsistency as send me an array of floats, "Nah, just send me a string"

Reply:
OK, we should find the path that is sending the `VALUE_EMBEDDED_OBJECT` and force it to send a string then